app/src/androidTest/java/org/simple/clinic/monthlyReports/questionnaire/QuestionnaireRepositoryAndroidTest.kt

```diff
@@ -0,0 +1,81 @@
+package org.simple.clinic.monthlyReports.questionnaire
+
+import com.google.common.truth.Truth
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.simple.clinic.AppDatabase
+import org.simple.clinic.TestClinicApp
+import org.simple.clinic.rules.SaveDatabaseRule
+import org.simple.sharedTestCode.TestData
+import org.simple.sharedTestCode.util.Rules
+import java.util.UUID
+import javax.inject.Inject
+
+class QuestionnaireRepositoryAndroidTest {
+
+  @Inject
+  lateinit var database: AppDatabase
+
+  @Inject
+  lateinit var questionnaireRepository: QuestionnaireRepository
+
+  @get:Rule
+  val rules: RuleChain = Rules
+      .global()
+      .around(SaveDatabaseRule())
+
+  @Before
+  fun setUp() {
+    TestClinicApp.appComponent().inject(this)
+  }
+
+  @Test
+  fun saving_questionnaires_should_work_correctly() {
+    // given
+    val questionnaires = listOf(
+        TestData.questionnaire(
+            uuid = UUID.fromString("ef5b7656-a6df-459c-a5b0-80d100721597"),
+            questionnaireType = MonthlyScreeningReports,
+            layout = "Test Questionnaire 1"
```

Comment:
It's usually good to have a accurate sample dataset here to indicate what users can expect. For example instead of a test string, we can include a test JSON representing the layout as would in prod.